packages/angular_devkit/build_angular/src/angular-cli-files/utilities/service-worker/index.ts

```diff
@@ -21,7 +20,7 @@ import { Observable, merge, of, from } from "rxjs";
 export const NEW_SW_VERSION = '5.0.0-rc.0';
 
 
-class CliFilesystem implements Filesystem {
```

Comment:
This doesn't seem to be a good change, since it will cause future changes to not respect the `Filesystem` constraints. 

Can you give some context about why this is needed?

Reply:
`@angular/service-worker` was an implicit dependency but due to the previous root package structure it was not caught.  This was already problematic as the version of the library directly imported can potentially differ from the version actual used and executed further in the file.  That version is required from the project itself whereas this one relies solely on node resolution.

Reply:
We can still import the type, and use `// tslint:disable-line:no-implicit-dependencies`. This way we don't lose the type safety while building.

Do do this for TSLint in `build-angular` for instance.

Reply:
True.  Also cleaned up some of the lint issues and used the types throughout.